tests/framework/pvc_utils.go

```diff
@@ -0,0 +1,192 @@
+package framework
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	defaultPvcMountPath = "/pvc"
+	poll                = 2 * time.Second
+)
+
+// Creates a PVC in the passed in namespace from the passed in PersistentVolumeClaim definition.
+// An example of creating a PVC without annotations looks like this:
+// CreatePVCFromDefinition(client, namespace, NewPVCDefinition(name, size, nil))
+func (f *Framework) CreatePVCFromDefinition(namespace string, def *k8sv1.PersistentVolumeClaim) (*k8sv1.PersistentVolumeClaim, error) {
+	var pvc *k8sv1.PersistentVolumeClaim
+	err := wait.PollImmediate(poll, DefaultTimeout, func() (bool, error) {
+		var err error
+		pvc, err = f.K8sClient.CoreV1().PersistentVolumeClaims(namespace).Create(def)
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		return false, err
+	})
+	if err != nil {
+		return nil, err
+	}
+	return pvc, nil
+}
+
+// Delete the passed in PVC
+func (f *Framework) DeletePVC(pvc *k8sv1.PersistentVolumeClaim) error {
+	err := wait.PollImmediate(poll, DefaultTimeout, func() (bool, error) {
+		err := f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.GetName()).Delete(pvc.GetName(), nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
+	})
+	return err
+}
+
+// Creates a PVC definition using the passed in name and requested size.
+// You can use the following annotation keys to request an import or clone. The values are defined in the controller package
+// AnnEndpoint
+// AnnSecret
+// AnnCloneRequest
+// You can also pass in any label you want.
+func NewPVCDefinition(pvcName string, size string, annotations, labels map[string]string) *k8sv1.PersistentVolumeClaim {
+	return &k8sv1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        pvcName,
+			Annotations: annotations,
+			Labels:      labels,
+		},
+		Spec: k8sv1.PersistentVolumeClaimSpec{
+			AccessModes: []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce},
+			Resources: k8sv1.ResourceRequirements{
+				Requests: k8sv1.ResourceList{
+					k8sv1.ResourceName(k8sv1.ResourceStorage): resource.MustParse(size),
+				},
+			},
+		},
+	}
+}
+
+// Wait for the PVC to be in a particular phase (Pending, Bound, or Lost)
+func (f *Framework) WaitForPersistentVolumeClaimPhase(phase k8sv1.PersistentVolumeClaimPhase, pvcName string) error {
+	wait.PollImmediate(poll, DefaultTimeout, func() (bool, error) {
+		pvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.GetName()).Get(pvcName, metav1.GetOptions{})
```

Comment:
why assume here that the ns is the one defined in the Framework. What if it's the cloner's ns2? I'd like to see consistent use of ns. Now it's sometimes it's passed, other times used from the Framework.

Reply:
You are right, I will make namespace a parameter to all the functions, for the reason that we might not be using the Namespace in the framework for that particular operation.